### ISSUE TITLE
Procedures subject_procedures and specimen_procedures may need slight modification

### ISSUE THREAD

**Issue by jtyoung84** [AllenNeuralDynamics/aind-data-schema#662]
**Describe the bug**
When running the unit tests in aind-metadata-service, the following warning shows up when serializing some of the models to json:

```
pydantic/main.py:308: UserWarning: Pydantic serializer warnings:
  Expected `Union[Craniotomy, FiberImplant, Headframe, IntraCerebellarVentricleInjection, IntraCisternalMagnaInjection, IontophoresisInjection, NanojectInjection, Perfusion, RetroOrbitalInjection, TrainingProtocol, WaterRestriction, SubjectProcedure]` but got `Headframe` - serialized value may not be as expected
```

**To Reproduce**
1. Create a Procedures model with a Headframe
2. Serialize it to json
3. Notice warnings

**Expected behavior**
There shouldn't be warnings that show up

**Additional context**
I think the root cause is that this
```
subject_procedures: Annotated[
        List[
            Union[
                Craniotomy,
                FiberImplant,
                Headframe,
                IntraCerebellarVentricleInjection,
                IntraCisternalMagnaInjection,
                IontophoresisInjection,
                NanojectInjection,
                Perfusion,
                RetroOrbitalInjection,
                TrainingProtocol,
                WaterRestriction,
                SubjectProcedure,
            ]
        ],
        Field(title="Subject Procedures", discriminator="procedure_type"),
    ] = []
```
needs to look like:
```
SUBJECT_PROCEDURES = Annotated[
  Union[
                Craniotomy,
                FiberImplant,
                Headframe,
                IntraCerebellarVentricleInjection,
                IntraCisternalMagnaInjection,
                IontophoresisInjection,
                NanojectInjection,
                Perfusion,
                RetroOrbitalInjection,
                TrainingProtocol,
                WaterRestriction,
                SubjectProcedure,
  ], Field(discriminator="procedure_type")]

subject_procedures: List[SUBJECT_PROCEDURES] = Field([], title="Subject Procedures")
```

**Comment by jtyoung84** [comment]
We should still implement the above suggestion because it fixes another issue that was raised. However, after running a few more tests, it seems that there are some serialization issues when an invalid class is constructed and serialized to json. The following still emits a serialization regardless:
```
h = Headframe.model_construct()
f = FiberImplant.model_construct()
subject_procedures = [h, f]
p = Procedures(subject_id="12345", subject_procedures = subject_procedures)
p.model_dump_json()

UserWarning: Pydantic serializer warnings:
  Expected `Union[Craniotomy, FiberImplant, Headframe, IntraCerebellarVentricleInjection, IntraCisternalMagnaInjection, IontophoresisInjection, NanojectInjection, Perfusion, RetroOrbitalInjection, TrainingProtocol, WaterRestriction, SubjectProcedure]` but got `Headframe` - serialized value may not be as expected
  Expected `Union[Craniotomy, FiberImplant, Headframe, IntraCerebellarVentricleInjection, IntraCisternalMagnaInjection, IontophoresisInjection, NanojectInjection, Perfusion, RetroOrbitalInjection, TrainingProtocol, WaterRestriction, SubjectProcedure]` but got `FiberImplant` - serialized value may not be as expected
```
